service/src/main/java/com/derpgroup/quip/manager/QuipManager.java

```diff
@@ -272,125 +297,129 @@ protected void doHelpRequest(Map<String,String> messageMap, SsmlDocumentBuilder
-    QuipMetadata metadata = (QuipMetadata) voiceInput.getMetadata();
-    doDefaultRequest(messageMap, builder, metadata);
+  protected void doHelloRequest(VoiceInput voiceInput, ServiceOutput serviceOutput) {
+    doDefaultRequest(voiceInput, serviceOutput);
   }
 
-  public void doDefaultRequest(Map<String, String> messageMap, SsmlDocumentBuilder builder, QuipMetadata metadata) {
-    String bot = metadata.getBot();
+  public void doDefaultRequest(VoiceInput voiceInput, ServiceOutput serviceOutput) {
+
+    QuipMetadata inputMetadata = (QuipMetadata) voiceInput.getMetadata();
+    String bot = inputMetadata.getBot();
     if(bot == null){
-      builder.text("I don't know how to handle requests for unnamed bots.");
+      String response = "I don't know how to handle requests for unnamed bots.";
+      serviceOutput.getVoiceOutput().setPlaintext(response);
+      serviceOutput.getVoiceOutput().setSsmltext(response);
+      serviceOutput.getVisualOutput().setTitle(response);
+      serviceOutput.getVisualOutput().setText(response);
       return;
     }
     switch (bot) {
     case "complibot":
-      doComplimentRequest(messageMap, builder, metadata);
+      doComplimentRequest(voiceInput, serviceOutput);
       break;
     case "insultibot":
-      doInsultRequest(messageMap, builder, metadata);
+      doInsultRequest(voiceInput, serviceOutput);
       break;
     default:
-      builder.text("I don't know how to handle requests for the bot named '" + bot + "'.");
+      String response = "I don't know how to handle requests for the bot named '" + bot + "'.";
+      serviceOutput.getVoiceOutput().setPlaintext(response);
+      serviceOutput.getVoiceOutput().setSsmltext(response);
+      serviceOutput.getVisualOutput().setTitle(response);
+      serviceOutput.getVisualOutput().setText(response);
       break;
     }
   }
 
   @Override
-  protected void doGoodbyeRequest(VoiceInput voiceInput,
-      SsmlDocumentBuilder builder) {
+  protected void doGoodbyeRequest(VoiceInput voiceInput, ServiceOutput serviceOutput) {
   }
 
   @Override
-  protected void doConversationRequest(VoiceInput voiceInput,
-      SsmlDocumentBuilder builder) throws DerpwizardException {
-
-    Map<String,String> messageMap = voiceInput.getMessageAsMap();
-    QuipMetadata metadata = (QuipMetadata) voiceInput.getMetadata();
-    String messageSubject = voiceInput.getMessageSubject();
-
-    switchOnSubject(messageSubject, messageMap, builder, metadata);
+  protected void doConversationRequest(VoiceInput voiceInput, ServiceOutput serviceOutput) throws DerpwizardException {
+    switchOnSubject(voiceInput.getMessageSubject(), voiceInput, serviceOutput);
   }
 
-  public void switchOnSubject(String messageSubject, Map<String,String> messageMap, SsmlDocumentBuilder builder, QuipMetadata metadata) throws DerpwizardException {
+  public void switchOnSubject(String messageSubject, VoiceInput voiceInput, ServiceOutput serviceOutput) throws DerpwizardException {
+    
```

Comment:
Iirc there were issues with some of our meta requests like "another"  when we had voice input as a parameter to this method